mkosi/config.py

```diff
@@ -353,6 +353,30 @@ def supports_smbios(self, firmware: QemuFirmware) -> bool:
     def supports_fw_cfg(self) -> bool:
         return self in (Architecture.x86, Architecture.x86_64, Architecture.arm, Architecture.arm64)
 
+    def default_qemu_machine(self) -> str:
+        m = {
+            Architecture.x86      : "q35",
+            Architecture.x86_64   : "q35",
+            Architecture.arm      : "virt",
+            Architecture.arm64    : "virt",
+            Architecture.s390     : "s390-ccw-virtio",
+            Architecture.s390x    : "s390-ccw-virtio",
+            Architecture.ppc      : "pseries",
+            Architecture.ppc64    : "pseries",
+            Architecture.ppc64_le : "pseries",
+        }
+
+        if self not in m:
+            die(f"No qemu machine defined for architecture {self}")
```

Comment:
Before, this defaulted to `q35`, while now it dies immediately. I assume the change is because it before qemu would have thrown an error instead. Explaining the reasoning behind this change in the commit message would be nice.